pkg/apis/k0s/v1beta1/clusterconfig_types.go

```diff
@@ -332,6 +336,64 @@ func (s *ClusterSpec) Validate() (errs []error) {
 		errs = append(errs, err)
 	}
 
+	for _, err := range s.validateKine(online) {
+		errs = append(errs, err)
+	}
+
+	return
+}
+
+func (s *ClusterSpec) validateKine(online bool) (errs field.ErrorList) {
+	if s.Storage.Type != KineStorageType {
+		return
+	}
+
+	if s.Storage.Kine == nil {
+		errs = append(errs, field.NotFound(field.NewPath("storage", "kine"), "missing"))
+		return
+	}
+
+	dataSource := s.Storage.Kine.DataSource
+
+	if dataSource == "" {
+		// sqlite - defaults
+		return
+	}
+
+	if !online {
+		return
+	}
+
+	path := field.NewPath("storage", "kine", "datasource")
+
+	u, err := url.Parse(dataSource)
```

Comment:
I have a vague memory that there were some weird edge cases when parsing the Kine datasource using url.Parse, but I may be mistaken. Maybe Windows related? :thinking:

_digging_  This one seems related: ac89d84f93f91003dd921b97794c92131de724aa.

We need to double-check this, especially for Windows paths.